### DISCUSSION TITLE
Laser out of the pre-defined boundries

### DISCUSSION THREAD

**Discussion by MaiLes839** [arkypita/LaserGRBL#1145] [Q&A]
Hi, when I'm adding a new image for engraving, I'm adding the image size, then running the "Boundary" button (see image below) to make sure the boundaries are OK. It seems to be perfect, but when I start the engraving job, the laser tries to travel beyond these boundaries (it get stuck since it is beyond the machine's engraving area. 
What am I doing wrong? something with the configuration? I gave $130 and $131 for the surface area of the machine...
Any help would be much appreciated! thanks
![Image 18](https://user-images.githubusercontent.com/72291216/96833636-769b7600-1449-11eb-948e-16d103f46134.png)
![Image 19](https://user-images.githubusercontent.com/72291216/96833829-ae0a2280-1449-11eb-9806-cb81045e12db.png)

**Comment by gmmanonymus111** [comment]
Could you try it with a smaller picture, and share the generated GCode with us, if the problem occurs? (`File`->`Quick Save`, then you can attach the file here)
When (after) it hit the limits, does limit switches shut your machine off, or does it try to move in other directions after a while?
What kind of engraver do you have?
EDIT: When it is stuck, coulds you send a screenshot of your LaserGRBL Command output window? (This one:)
![image](https://user-images.githubusercontent.com/9658243/96861713-c471a780-1464-11eb-8fd4-0353193b0f1c.png)

**Comment by MaiLes839** [comment]
Thanks Alex!

_When (after) it hit the limits, does limit switches shut your machine off, or does it try to move in other directions after a while?_
Neither. It tries to continue in the same direction (but of course can't. It only stops when I click the stop button manually.

_What kind of engraver do you have?_
It's called STUMP:
![image](https://user-images.githubusercontent.com/72291216/96870885-36082080-147a-11eb-8549-a22ff94641fa.png)


_EDIT: When it is stuck, could you send a screenshot of your LaserGRBL Command output window?_ 
![image](https://user-images.githubusercontent.com/72291216/96872212-02c69100-147c-11eb-8233-7bade5a0b7fc.png)

It stops only when I click the "STOP" BUTTONbutton:

![image](https://user-images.githubusercontent.com/72291216/96872292-1eca3280-147c-11eb-9b06-73a8389401fd.png)

**Comment by gmmanonymus111** [comment]
**“error:25” : _(“A G-code word was repeated in the block.”)**
This can be either because LaserGRBL sends something it shouldn't (I doubt it), or by electrical noise during communication (more likely). It is hard to tell without a sample GCode.
In the meantime, you can read the section "_Weird result, hang, reset or errors._" of the faq [here](https://lasergrbl.com/faq/)
EDIT: You could try a [check-mode](https://github.com/gnea/grbl/wiki/Grbl-v1.1-Commands#c---check-gcode-mode) run to see if the engraving finishes that way.

**Comment by StuartB4** [comment]
Looks like your settings are wrong for $130 and $131. X axis is 130. From the picture you posted above the X axis should be 200. and the Y 170

**Comment by StuartB4** [comment]
Unless they do it the other way round on that machine. Using the bar that the laser slides on as the Y axis.

**Comment by MaiLes839** [comment]
Alex - thanks I will try what you said.
Stuart - I think the settings are OK because when I do a test run for the boundaries before starting the job (see image below) it marks the boundaries perfectly, but I will try to change the settings and see if it helps... Thanks!

**Comment by StuartB4** [comment]
Ok.
